core/src/main/java/com/novoda/merlin/service/HostPinger.java

```diff
@@ -21,29 +16,22 @@
 
     }
 
-    static HostPinger withDefaultEndpointValidation(PingerCallback pingerCallback) {
-        Logger.d("Host address not set, using Merlin default: " + Merlin.DEFAULT_ENDPOINT);
-        PingTaskFactory pingTaskFactory = new PingTaskFactory(pingerCallback, new ResponseCodeFetcher(), new DefaultEndpointResponseCodeValidator());
-        return new HostPinger(pingerCallback, Endpoint.defaultEndpoint(), pingTaskFactory);
-    }
-
-    static HostPinger withCustomEndpointAndValidation(PingerCallback pingerCallback, Endpoint hostAddress, ResponseCodeValidator validator) {
-        PingTaskFactory pingTaskFactory = new PingTaskFactory(pingerCallback, new ResponseCodeFetcher(), validator);
-        return new HostPinger(pingerCallback, hostAddress, pingTaskFactory);
+    static HostPinger withCustomEndpointAndValidation(Endpoint hostAddress, ResponseCodeValidator validator) {
+        PingTaskFactory pingTaskFactory = new PingTaskFactory(new ResponseCodeFetcher(), validator);
+        return new HostPinger(hostAddress, pingTaskFactory);
     }
 
-    HostPinger(PingerCallback pingerCallback, Endpoint endpoint, PingTaskFactory pingTaskFactory) {
-        this.pingerCallback = pingerCallback;
+    HostPinger(Endpoint endpoint, PingTaskFactory pingTaskFactory) {
         this.endpoint = endpoint;
         this.pingTaskFactory = pingTaskFactory;
     }
 
-    void ping() {
-        PingTask pingTask = pingTaskFactory.create(endpoint);
+    void ping(PingerCallback pingerCallback) {
```

Comment:
Dependency needed to be inverted to allow the creation of a `HostPinger` in the `MerlinBuilder` but keeping the callback in the `MerlinService`.